text/0000-ember-new-lang.md

```diff
@@ -0,0 +1,239 @@
+- Correct dictionary lookups for translations
+- Assisting search engines
+- Improving typography in certain situations
+
+Accordingly, while the primary motivation of this RFC is to address an unresolved digital accessibility issue in Ember, it is expected that a successful implementation of the proposed `--lang` flag solution will provide additional, non-accessibility-related improvements to the baseline quality of new Ember applications.
+
+## Detailed design
+
+Link to [candidate implementation](https://github.com/josephdsumner/ember-cli/compare/master...ember-new-lang-base).
+
+We have explicitly chosen `--lang` as the flag (vs `--language`) for consistency with the HTML attribute itself. 
+
+```bash
+ember new my-app --lang en-US
+# -l is also a valid alias
+```
+
+The above ember-cli command will result in the following `index.html` header change.
+
+```html
+<html lang="en-US">
+```
+
+The flag is added to relevant ember-cli help commands, such as the following:
+
+```bash
+ember help new
+ember new <app-name> <options...>
+  ...
+  --lang (String) (Default: "") Sets the base human language of the application via index.html
+    alias: -l <value>
+```
+
+### Misusage and Error Handling
+
+
+Broadly, incorrect usage of the `--lang` flag covers three use-case categories:
+
+- The flag has been specified with a programming language as the argument
+- The flag has been specified with no argument
+- The flag has been specified with an invalid language code
+
+This RFC proposes that these cases cause the build process to halt with a revelant error and help message as opposed to simply reverting to the default value and reporting a message. **The rationale for the recommendation to halt instead of just report is that in these all of these cases, the user has explicitly typed `--lang` into their CLI tool. This is an unambiguous declaration of intent by the user to use the `--lang` flag correctly.**
+
+
+
+#### Invalid Language Codes
+
+Language codes are verified against [is-language-code](https://www.npmjs.com/package/is-language-code). (see [examples of valid ISO country codes](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes))
+
+If an invalid language code is given such as `--lang en-UK` the indended output should be a shell error that will halt the build.
+
+```bash
+ember new my-app --lang en-UK
+Unrecognized language subtag, "uk".
+```
+
+#### No argument passed
+Additionally, if the `--lang` flag is used but no argument is defined, e.g., `ember new my-app --lang`, the build will also be halted. An error message and usage info should be shown in these cases:
+
+```bash
+# Detect trailing option (declared)
+ember new my-app --lang --skip-git
+
+An error with the `--lang` flag returned the following message:
+  Detected lang specification starting with command flag `-`.
+  Is `--skip-git` meant to be an ember-cli command option?
+  This issue is likely caused by using the `--lang` flag without a specification.
```

Comment:
Again I think this shouldn't normative in the RFC anyway, but IMO this is working around a bug in the argument parsing, so exposing this much information about the limitation/bug/workaround in our argument parsing is more confusing to the user than it is helpful. IMO, I think both of these cases should show something short like "The `--lang` flag requires a valid language code, see `ember <command> help for more information.". In the second example, the `--disable-analytics` bit is extra confusing, because it's not part of what the user typed and definitely a quirk of our internal argument parsing implementation.

Concretely, if the parsed value starts with a dash, I think it's just a bug in the parsing and we can show a message about the missing language code. Otherwise, we just validate against the ISO list.

Reply:
+1. I think a more consistent way to do this is when parse fails, print a message and either a descriptive help message with what the flag is for, or a link to a URL that has "common misunderstandings" if it is necessary. Trying to guess user intention is probably not going to scale very well.